docs/DeveloperGuide.adoc

```diff
@@ -345,6 +345,61 @@ image::UndoRedoActivityDiagram.png[width="200"]
 **Cons:** Requires dealing with commands that have already been undone: We must remember to skip these commands. Violates Single Responsibility Principle and Separation of Concerns as `HistoryManager` now needs to do two different things. +
 // end::undoredo[]
 
+// tag::findClose[]
+=== Fuzzy finding mechanism
+The fuzzy finding mechanism is powered by https://en.wikipedia.org/wiki/Levenshtein_distance[Levenshtein distance],
+a string heuristic for measuring the difference between two character sequences. It can be informally viewed as the number
+of _hops_ that are required to get from a string to another, where _hops_ are insertions, deletions and substitution
+operations.
+
+image::FuzzyFindActivityDiagram.png[]
+
+The activity diagram for the fuzzy finding can be loosely described as seen in the figure above.
+
+1. A user sends a find request
+2. Application looks at all contacts currently in the database
+3. If the search parameters loosely match any of the name words of the current contact, add it into the list view.
+4. Otherwise, if it is an exact match on any of the name words of the current contact, add it into the list view.
+5. Otherwise, if it matches any other conditions specified under the find command, add it into the list view.
+
+[NOTE]
+If a contact's name word has a shorter length than the specified global minimum, the condition automatically triggers to false
+and starts searching for exact matches.
+
+The settings for the loose matching can be found as a static constant of the Person class.
+
+[source,java]
+----
+public class Person implements ReadOnlyPerson {
+    private static final int FIND_NAME_GLOBAL_TOLERANCE = 4;
+    private static final int FIND_NAME_DISTANCE_TOLERANCE = 2;
+    ...
+}
+----
+
+`FIND_NAME_GLOBAL_TOLERANCE` is the global minimum distance of the currently examined name word for the loose matching to execute.
+`FIND_NAME_DISTANCE_TOLERANCE` is the maximum levenshtein distance from the currently examined search parameter
+to the currently examined name word in which the contact will be added.
+
+==== Design Considerations
+
+**Aspect:** Length of name word +
+**Alternative 1 (current choice):** Use a global minimum for defining when the fuzzy finding should execute. +
+**Pros:** Name words have higher minimum hop to length ratio. +
+**Cons:** User might need fuzzy finding on name words with character length lesser than the global minimum. +
```

Comment:
Add alternative 2 to compare and contrast decisions, strengthening rationale for current choice?

Reply:
Can't think of alternative for this one. Coz you'd need to design some kind of algorithm that does it better. Have any suggestions?

Reply:
Hmm what if we also execute fuzzy finding for common name words that are shorter than the global minimum? Then users may find the search more useful but implementation requires more work.

Reply:
@jo-lyn that's the problem we're trying to solve.